packages/worker/src/helpers/scanner.ts

```diff
@@ -1,22 +1,45 @@
-import { File, ipfs } from '@aragonone/ipfs-background-service-shared'
+import { FileMeta, ipfs, etherscan } from '@aragonone/ipfs-background-service-shared'
+import metrics from './metrics-reporter'
 
 export default async function scan() {
-  const files = await File.findUnverified()
+  const files = await FileMeta.findUnverified()
   for (const file of files) {
-    const { cid } = file
+    const { cid, owner } = file
     if (file.isExpired()) {
       await file.del()
-      await ipfs.del(String(cid))
+      await ipfs.del(cid!)
+      metrics.fileExpired()
+      console.log(`Expired file ${cid} deleted`)
     }
     else {
-      /*
-      to-do: scan blockchain:
-      - get lastScannedBlock from db
-      - get owner transaction receipts since lastScannedBlock
-      - get event data and look for cid
-      - set verified = true, expiresAt = null if cid is found
-      - update lastScannedBlock to current block
-      */
+      console.log(`Scanning owner address ${owner} for file ${cid}`)
+      await tryVerifyCid(file)
+      metrics.fileScanned()
     }
   }
 }
+
+async function tryVerifyCid(file: FileMeta) {
+  const { cid, owner } = file
+  const startblock = file.lastScannedBlock! + 1
```

Comment:
We should probably scan more than one block per run, probably we can do a diff with the current block number and cap it to scan up to 10 blocks or so

Reply:
Oh this is just where I start the scan. `findIpfsCid` will scan as many as possible up to 10,000 query results :)